app/controllers/products_controller.rb

```diff
@@ -1,37 +1,34 @@
 class ProductsController < ApplicationController
-  before_action :require_login, except: [:index, :show, :merchant_products]
+  before_action :require_login, only: [:new, :create, :edit, :update, :destroy, :retire]
+  before_action :find_product,  only: [:show, :edit, :update, :destroy, :retire]
 
   def index
     @products = Product.active_product
   end
 
   def show
-    @product = Product.find(params[:id])
     @order_item = current_order.order_items.new
   end
 
   def new
-    @product = Product.new
+    user = User.find(params[:user_id])
+    @product = Product.new(user_id: user.id)
```

Comment:
It seems like with the way the code is written, you could skip line 14, and plug the `params[:user_id]` directly into the product `new`. 
Although that is the case, I think the User lookup prior to just using the Id is good, but I would prefer if you did some sort of error handling on whether or not that user is found before plugging it into the product creation and then it would seem more worthwhile.